src/regression.js

```diff
@@ -30,12 +30,26 @@
   /**
    * Determine the coefficient of determination (r^2) of a fit from the observations and predictions.
    *
+   * @param {Array<Array<number>>} data - Pairs of observed x-y values, including nulls
+   * @param {Array<Array<number>>} results - Pairs of observed predicted x-y values, including null predictions
+   *
    * @param {Array<Array<number>>} observations - Pairs of observed x-y values
    * @param {Array<Array<number>>} predictions - Pairs of observed predicted x-y values
    *
    * @return {number} - The r^2 value, or NaN if one cannot be calculated.
    */
-  function determinationCoefficient(observations, predictions) {
+  function determinationCoefficient(data, results) {
+    var observations = [];
+    var predictions = [];
+
+    // Remove predictive points for accurate fit
+    data.forEach( function (d, i) {
```

Comment:
I don't understand this change. You filter for null below when calculating a value which requires a length calculation. Why filter null here as well? What if you want to retain the 0-fill behavior?

Seems like there are a few options:

1. Make the entire library filter out null
2. Retain the original behavior and add a flag for conditionally filtering null
3. Don't change the library at all and filter null at the input

It feels like this change moves us further away from the base library. Why not just create a helper method that does the null filtering then passes it to the regression library?